:book: Updates to vSphere Quick Start based on upcoming v0.5.2 changes

andrewsykim opened the pull request:
Signed-off-by: Andrew Sy Kim <kiman@vmware.com>

**What this PR does / why we need it**:
Updates CAPV's quick start guide based on the upcoming v0.5.2 changes. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

akutz commented on docs/book/src/user/quick-start.md:
````diff
@@ -434,82 +474,84 @@ spec:
-These examples include environment variables that you should substitute before creating the resources.
+This quick start assumes the following vSphere environment which you should replace based on your own environment:
+
+```
````
Hi @andrewsykim,

Please see my earlier comment about the formatting of these properties.

andrewsykim commented on docs/book/src/user/quick-start.md:
```diff
@@ -73,14 +73,36 @@ curl -L {{#releaselink gomodule:"sigs.k8s.io/cluster-api-provider-aws" asset:"in
+  name: capv-manager-bootstrap-credentials
+  namespace: capv-system
+type: Opaque
+data:
```
Thanks, meant to use stringData but forgot.

ncdc commented:
@andrewsykim do you have an ETA for finishing up the changes requested?

andrewsykim commented:
@ncdc maybe in a week? I actually haven't addressed the comments because the updates are blocked on changes in CAPV - https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/issues/609 and https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/issues/608

ncdc commented:
Thanks! I didn't realize there were still CAPV changes in flight. Really just checking in to make sure this doesn't get lost.

andrewsykim commented:
Thanks for checking up! It's one of the blocking items for v0.5.2 so I'll make sure to follow-up on this soon :)

andrewsykim commented on docs/book/src/user/quick-start.md:
````diff
@@ -73,14 +73,36 @@ curl -L {{#releaselink gomodule:"sigs.k8s.io/cluster-api-provider-aws" asset:"in
 ```bash
-kubectl create -f {{#releaselink gomodule:"sigs.k8s.io/cluster-api-provider-vsphere" asset:"infrastructure-components.yaml" version:"0.5.x"}}
+# Upload vCenter credentials as a Kubernetes secret
+$ cat <<EOF | kubectl apply -f -
````
The infrastructure-components deployments would hang without the secret, didn't want to leave room for confusion because a user thinks applying infrastrucutre-components.yaml should just work.

andrewsykim commented:
Ready for another review, PTAL

akutz commented:
/lgtm

vincepri commented:
Should we point the branch to `release-0.2` or backport later?

andrewsykim commented:
@vincepri I'm in favor of back porting since it doesn't seem like the other docs are still (mostly?) v1alpha2 references. Will defer to you though since I'm not familiar with the state of branches here

ncdc commented:
👍 to backporting to `release-0.2`

vincepri commented:
/retitle :book: Updates to vSphere Quick Start based on upcoming v0.5.2 changes
